src/engine/Execute.v3

```diff
@@ -25,7 +25,7 @@ component Execute {
 }
 
 type StackSwitchInfo {
-	case Suspend(top: WasmStack, tagdef: Tag, vals: Array<Value>) {}
+	case Suspend(top: WasmStack, tagdef: Tag, vals: Array<Value>, is_switch: bool) {}
```

Comment:
Hmm, it might be cleaner to introduce a new ADT case here, rather than a bool switch. Wdyt?

Reply:
I think this is a good idea. I just finished the V3Int implementation of `switch` and currently I'm abusing the stack to pass continuation values around. With a new `switch` case this would be much cleaner.